### DISCUSSION TITLE
Regarding how to access window controls from another window

### DISCUSSION THREAD

**Discussion by IronEaterBeast** [AvaloniaUI/Avalonia#15439] [Q&A]
I have a window called test with a textbox named textBox. I created a new instance of the test window in another window using: test testWindow = new test ();

Now, I want to access this textbox using "testWindow.textBox", but it cannot be found. Is this method of access incorrect or is the textBox a private element that cannot be accessed externally? Can someone explain the reasoning behind this?

**Comment by Pratham95VP** [comment]
```
public partial class Test : Window
{
    public TextBox textBox { get; set; }

    public Test()
    {
        InitializeComponent();
        this.FindControl<TextBox>("textBox").Name = "textBox";
    }
}
```
```
 Test testWindow = new Test();
TextBox tb = testWindow.textBox;
```
 When you create a user interface element in AXAML, by default, its access modifier is set to private. This means that the element can only be accessed within the same class where it is declared.

**Comment by timunie** [comment]
I highly reconmend to follow MVVM pattern and use Bindings

**Comment by maxkatz6** [comment]
You also can use FieldModifier on your element. This way you don't need `this.FindControl` and changes in c# code-behind:

```xaml
<TextBox x:Name="textBox" x:FieldModifier="Public />
```

**Reply by IronEaterBeast** [reply]
Cool, the simplest way.
